src/lintsampler/lintsampler.py

```diff
@@ -0,0 +1,547 @@
+            # tally the number of gridpoints
+            if ngrid == None:
+                ngridpoints = np.prod(self.edgedims)
+
+                # create the flattened grid for evaluation
+                edgegrid = np.stack(np.meshgrid(self.edgearrays, indexing='ij'), axis=-1).reshape(ngridpoints, self.dim)
+            else:
+                ngridpoints = self.ngridentries[ngrid]
+
+                # create the flattened grid for evaluation
+                edgegrid = np.stack(np.meshgrid((self.edgearrays[ngrid]), indexing='ij'), axis=-1).reshape(ngridpoints, self.dim)
+
+
+            # call the pdf
+            evalfgrid = np.zeros(ngridpoints)
+            for gridpoint in range(0,ngridpoints):
+                evalfgrid[gridpoint] = self.pdf(edgegrid[gridpoint],*funcargs)
+            
+            # reshape back to the grid
+            if ngrid == None:
+                evalfgrid = evalfgrid.reshape(self.gridshape[0:self.dim])
+            else:
+                evalfgrid = evalfgrid.reshape(self.gridshape[ngrid][0:self.dim])
+
+        return evalfgrid
+
+    def sample(self, N_samples=None, funcargs=()):
+        """Draw samples from the pdf on the constructed grid(s).
+        
+        This function then draws a sample (or N samples) from the specified grid. It first chooses a cell 
+        (or N cells with replacement), weighting them by their mass (estimated by the trapezoid rule) 
+        then samples from k-linear interpolant within the chosen cell(s).
+        
+        Parameters
+        ----------
+        self : LintSampler
+            The LintSampler instance.
+        N_samples : {None, int}, optional
+            Number of samples to draw. Default is None, in which case a single
+            sample is drawn.
+        funcargs : optional
+            A tuple of optional arguments to be passed directly to the pdf function call.
+
+        Returns
+        -------
+        X : scalar, 1D array (length k OR N_samples) or 2D array (N_samples, k)
+            Sample(s) from linear interpolant. Scalar if single sample (i.e.,
+            N_samples is None) in 1D. 1D array if single sample in k-D OR multiple
+            samples in 1D. 2D array if multiple samples in k-D.
+        
+        
+        """
+        
+        # check that the function can be evaluated at the passed-in points (i.e. needs to take the dimensions as an argument)
+
+        if self.eval_type == 'gridsample':
+
+            if self.dim > 1:
+                
+                # evaluate the pdf
+                evalf = self._evaluate_gridded_pdf(funcargs)
+
+                # call the gridded sampler
+                X = _gridsample(*self.edgearrays,f=evalf,N_samples=N_samples,seed=self.rng)
+
+            else: 
+
+                # evaluate the pdf
+                evalf = self._evaluate_gridded_pdf(funcargs)
+
+                # call the gridded sampler
+                X = _gridsample(self.edgearrays,f=evalf,N_samples=N_samples,seed=self.rng)
```

Comment:
Below (in `setgrid`), I make the suggestion of always having `edgearrays` as a list, even in the 1D case. Then, we don't need the two cases here.
```suggestion             
            # evaluate the pdf
            evalf = self._evaluate_gridded_pdf(funcargs)

            # call the gridded sampler
            X = _gridsample(*self.edgearrays,f=evalf,N_samples=N_samples,seed=self.rng)
```